### PR TITLE
feat(ST-38): develop and add dashboard page

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
## Description of Change

- Created a UserDashboard module
- Created a component called AllClassrooms
- Showed the empty classrooms components for teacher & student
- Made student and teacher components dynamic
- Added the dashboard page route after login and registration

## Associated Ticket Link(s)

- https://trello.com/c/HSnUUXum

## Related Pull Request(s)

- N/A

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/c302e087-be60-44c2-8d5d-491028b95bef

### CODE REVIEW

File: modules/UserDashboard/components/AllClassrooms/AllClassrooms.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import { FaPlus } from "react-icons/fa";
+import { useSelector } from "react-redux";
+
+import { TRootState } from "@/shared/redux/store";
+import { ERole } from "@/shared/typedefs";
+
+import classes from "./AllClassrooms.module.css";
+
+const AllClassrooms = () => {
+  const user = useSelector((state: TRootState) => state.authenticatedUser);
```
Comment (ishmam-iqbal-sazim) [inline modules/UserDashboard/components/AllClassrooms/AllClassrooms.tsx:11]:
```suggestion
  const user = useAppSelector(authenticatedUserSelector);
```

Should be typed. Look into the template for useAppSelector implementation.

Should also add a const for whatever you're trying to fetch from the reducer slice and import it here. authenticatedUserSelector in this case

File: modules/components/Navbar/NavigationBar.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const NavigationBar = () => {
+  const role = useSelector((state: TRootState) => state.authenticatedUser.claim);
+  return (
+    <Box h={"9vh"} bg={"#011733"} className={classes["navlinks"]}>
+      <Text>Superteacher</Text>
+      <Flex justify={"center"} align={"center"} gap={15}>
+        <Text className={classes["dashboard"]}>Dashboard</Text>
+        {role === ERole.TEACHER && <FaPlus />}
+        <Button variant="outline" c={"white"} sx={{ borderColor: "white" }}>
+          Super
```
Comment (ishmam-iqbal-sazim) [inline modules/components/Navbar/NavigationBar.tsx:21]: Is this a placeholder?

Comment (faysal-alam-sazim) [reply modules/components/Navbar/NavigationBar.tsx:21]: > Is this a placeholder?

Yes bhai.

File: modules/UserDashboard/components/AllClassrooms/AllClassrooms.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import { FaPlus } from "react-icons/fa";
+
+import { useAppSelector } from "@/shared/redux/hooks";
+import { authenticatedUserSelector } from "@/shared/redux/reducers/user.reducer";
+import { ERole } from "@/shared/typedefs";
+
+import classes from "./AllClassrooms.module.css";
+
+const AllClassrooms = () => {
+  const user = useAppSelector(authenticatedUserSelector);
```
Comment (ishmam-iqbal-sazim) [inline modules/UserDashboard/components/AllClassrooms/AllClassrooms.tsx:11]: Please add line break below

File: modules/UserDashboard/components/AllClassrooms/AllClassrooms.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  return (
+    <Box mih={"91vh"} pos={"relative"}>
+      {user.claim === ERole.TEACHER ? (
```
Comment (ishmam-iqbal-sazim) [inline modules/UserDashboard/components/AllClassrooms/AllClassrooms.tsx:14]:
```suggestion
      {user?.claim === ERole.TEACHER ? (
```

[NIT]: Best to use optional chaining. If the user logs out from this page, the user object would be empty and attempting to access user.claim would throw an error.

File: modules/components/Navbar/NavigationBar.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+import { useAppSelector } from "@/shared/redux/hooks";
+import { authenticatedUserSelector } from "@/shared/redux/reducers/user.reducer";
+import { ERole } from "@/shared/typedefs";
+
+import classes from "./NavigationBar.module.css";
+
+const NavigationBar = () => {
+  const user = useAppSelector(authenticatedUserSelector);
+  return (
```
Comment (ishmam-iqbal-sazim) [inline modules/components/Navbar/NavigationBar.tsx:14]: Please add line break above

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

File: modules/UserDashboard/components/AllClassrooms/AllClassrooms.tsx
Comment (ragib-hossain-sazim) [inline modules/UserDashboard/components/AllClassrooms/AllClassrooms.tsx:1]: [Nitpick] `AllClassrooms` looks like a container. Change it to `ClassroomContainer` and adjust the file folders accordingly.

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved Pending comments

**Pull request merged**